x-pack/platform/plugins/shared/fleet/server/services/agent_policy_create.ts

```diff
@@ -155,6 +155,18 @@ export async function createAgentPolicyWithPackages({
       newPolicy.is_default_fleet_server = true;
     }
   }
+
+  const withSysMonitoring = withSysMonitoringParams && !newPolicy.supports_agentless;
```

Comment:
Shouldn't we set the `monitoringEnabled:[]` from the UI instead? It's a bit misleading to have it set but not take effect in case of serverless.

Reply:
Having this server side, allow to make sure it's never created even if a user by mistake enable it through the API and avoid introducing a breaking change where the server will refuse those values for agentless, I agree it could be a little confusing, it's why I added some log, do you thing bumping the log level to `info` instead of `debug` could help to make this less confusing?

Reply:
Yeah it sounds good